tools/tplgtool.py

```diff
@@ -777,12 +784,73 @@ def get_tplg_paths(cmd_args):
             tplg_paths.append(f)
         return tplg_paths
 
+    # connect nodes
+    def connect(graph, head):
+        assert(head is not None) # make sure head is not None
+        if head["sink"] != None and type(head["sink"]) != list:
+            graph.edge(head["name"], head["sink"]["name"])
+            connect(graph, head["sink"])
+        elif head["sink"] != None:
+            for node in head["sink"]:
+                graph.edge(head["name"], node["name"])
+                connect(graph, node)
+
+    # traverse all nodes, and add them to graph
+    def init_node(graph, head):
+        # this function is used to deal with multiple connections.
+        def inner_init(graph, head):
+            if head["sink"] != None and type(head) != list:
+                graph.node(name=head["name"],)
+                init_node(graph, head["sink"])
+            elif head["sink"] != None:
+                for node in head["sink"]:
+                    init_node(graph, node,)
+            else :
+                graph.node(name=head["name"])
+
+        assert(head is not None) # make sure head is not None
+        if type(head) != list:
+            inner_init(graph, head)
+        else :
+            for subhead in head:
+                inner_init(graph, subhead)
+
     def dump_pcm_info(parsed_tplgs):
         for tplg in parsed_tplgs:
             formatter = TplgFormatter(tplg)
             formatter.format_pcm()
             print()
 
+    def dump_graph(parsed_tplgs, cmd_args):
+        try:
+            from graphviz import Digraph
+        except:
+            print("graphviz package not installed, please install with `sudo apt install python3-graphviz`")
+            sys.exit(1)
+
+        format = cmd_args['format'].strip()
+        dir = cmd_args['directory'].strip()
+
+        for tplg in parsed_tplgs:
+            # the last element in tplg is topology path
+            outfile = tplg[-1].split(sep='/')[-1].split('.')[0]
+            formatter = TplgFormatter(tplg)
+            head_list, _ = formatter._link_graph()
```

Comment:
please change `_link_graph` to `link_graph`

Reply:
done